core/model-api/src/main/java/org/eclipse/rdf4j/model/base/AbstractLiteral.java

```
@@ -200,6 +201,32 @@ private boolean equals(Optional<String> x, Optional<String> y) {
 		return px && py && x.get().equalsIgnoreCase(y.get()) || !px && !py;
 	}
 
+	public enum BaseDirection {
+		NONE(""),
+		LTR(Literal.LTR_SUFFIX),
+		RTL(Literal.RTL_SUFFIX);
+
+		private final String suffix;
+
+		BaseDirection(final String suffix) {
+			this.suffix = suffix;
+		}
+
+		public String getSuffix() {
+			return suffix;
+		}
+
+		public static BaseDirection fromString(String dir) {
+			if (dir == null || dir.isEmpty())
+				return NONE;
+			if (dir.equalsIgnoreCase(Literal.LTR_SUFFIX))
+				return LTR;
+			if (dir.equalsIgnoreCase(Literal.RTL_SUFFIX))
+				return RTL;
+			return NONE;
```

If suffix is not valid, it will ignore it. Do we want to throw an exception of silently ignore?

---

There is checking in RDFParserHelper that verifies that direction is either ltr or rtl, and throws a fatal RDFParseException if an invalid direction is read. Is that enough, or do we also want checking here which is called by the Literal constructors?

---

Creating an literal is a different path. IMO, users can just use the enum value. like 
```java
public Literal createLiteral(String label, String language, BaseDirection baseDirection)
```